semtypes/src/main/java/io/ballerina/types/subtypedata/XmlSubtype.java

```diff
@@ -80,7 +80,7 @@ public static SemType xmlSequence(SemType constituentType) {
         // It is a precondition that constituentType is a subtype of XML
         assert Core.isSubtypeSimple(constituentType, PredefinedType.XML);
 
-        if (constituentType == PredefinedType.NEVER) {
+        if (constituentType.equals(PredefinedType.NEVER)) {
```

Comment:
Shall we use `isNever` instead
```suggestion
        if (Core.isNever(constituentType)) {
```